src/v/cloud_storage/topic_mount_manifest.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "cloud_storage/base_manifest.h"
+#include "cloud_storage/fwd.h"
+#include "cloud_storage/types.h"
+#include "model/fundamental.h"
+#include "model/metadata.h"
+
+#include <seastar/core/future.hh>
+
+#include <optional>
+
+namespace cloud_storage {
+
+class topic_mount_manifest final : public base_manifest {
+public:
+    topic_mount_manifest(
+      const model::cluster_uuid& uuid,
+      const model::topic_namespace& tp_ns,
+      const std::optional<ss::sstring>& recovery_prefix = std::nullopt);
+
+    ss::future<> update(ss::input_stream<char> is) override;
+
+    ss::future<serialized_data_stream> serialize() const override;
+
+    ss::sstring get_recovery_prefix() const;
+
+    const model::cluster_uuid& get_uuid() const { return _uuid; }
+
+    const model::topic_namespace& get_tp_ns() const { return _tp_ns; }
+
+    /// Manifest object name in S3
+    remote_manifest_path
+    get_manifest_path(const remote_path_provider& path_provider) const;
+
+    manifest_type get_manifest_type() const override;
+
+private:
+    model::cluster_uuid _uuid;
+    model::topic_namespace _tp_ns;
+    std::optional<ss::sstring> _recovery_prefix;
```

Comment:
Ah looked again at the original proposal. My original thoughts were that the recovery prefix would pretty much be the cluster UUID, but I didn't want to leave it open in case we wanted to use some other unique identifying string. In practice today, I think the closest thing we have is the remote_label, which I suggest we use instead of a plain cluster_uuid. If so, I think we can remove this prefix.

Reply:
Removed `recovery_prefix` from `topic_mount_manifest` and updated manifest path to `migration/<source_label>/<namespace>/<topic>`.